frontend/src/components/ViewTrips/save-trip-modal.js

```diff
@@ -9,65 +9,120 @@ import { formatTripData } from '../Utils/filter-input.js';
 const db = app.firestore();
 
 /**
- * Returns a Form.Control element with input type 'text' and other fields
+ * Returns a Form.Control element with input type 'text' and other props
  * specified by the function parameters.
  *
- * @param {string} placeholder Text placehold in the form input
  * @param {React.RefObject} ref Ref attached to the value inputted in the form.
+ * @param {boolean} isAddTripForm True if form is adding new trip, false if
+ *     form is editting existing trip.
+ * @param {string} placeholder Placeholder text value in the form input.
+ * @param {!string=} defaultText Optional default text value in the form input.
  * @return {JSX.Element} The Form.Control element.
  */
-function createTextFormControl(placeholder, ref) {
+function createTextFormControl(ref, isAddTripForm,
+                                placeholder, defaultText = null) {
+  if (isAddTripForm) {
+    return (
+      <Form.Control
+        type='text'
+        placeholder={placeholder}
+        ref={ref}
+      />
+    );
+  }
   return (
     <Form.Control
-      type="text"
+      type='text'
       placeholder={placeholder}
+      defaultValue={defaultText}
       ref={ref}
     />
   );
 }
```

Comment:
Yes there are two. Only one executes based on what type of form this SaveTripModal is. I updated this in the most recent commit because I didn't really like this. Check it out and let me know if you have any questions.